### PR TITLE
Fix corkscrew

### LINKED ISSUE #273771: Build failure: corkscrew
Corkscrew is broken for `aarch64-darwin` on both `23.11` and unstable

### Steps To Reproduce

`nix build nixpkgs/nixos-23.11#corkscrew` and `nix build nixpkgs-unstable#corkscrew` fail with the same error (see below for build log).

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/5yfmrs1b6s2pg9612s06jxy9klcj7m92-corkscrew-2.0.tar.gz
source root is corkscrew-2.0
setting SOURCE_DATE_EPOCH to timestamp 998591634 of file corkscrew-2.0/README
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
configure flags: --prefix=/nix/store/2jylazj7dl6s089155y9b15qlqq26jck-corkscrew-2.0
creating cache ./config.cache
checking for a BSD compatible install... /nix/store/h0fgpyxfav7ybnw5sdg8jpyv9f95x4w0-coreutils-9.3/bin/install -c
checking whether build environment is sane... yes
checking whether make sets ${MAKE}... yes
checking for working aclocal... missing
checking for working autoconf... missing
checking for working automake... missing
checking for working autoheader... missing
checking for working makeinfo... missing
checking for gcc... clang
checking whether the C compiler (clang  ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.
```

### Notify maintainers

No maintainers listed

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
λ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.5.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.3`
 - nixpkgs: `/nix/store/mpsjbqbyzcdjbgf2phj0yik4nk8dp1xv-mbz4hixfgxq5b6vc0k3pp2iglcd4c353-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by JackSullivan**
## Description of changes

Updates the source of corkscrew from a dead link to a more recent github, and fixes it such that it compile and runs (tested on aarch64-darwin). Fixes #273771.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/corkscrew/default.nix [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
   version = "2.0";
-
-  src = fetchurl {
-    url = "http://agroman.net/corkscrew/corkscrew-${version}.tar.gz";
-    sha256 = "0d0fcbb41cba4a81c4ab494459472086f377f9edb78a2e2238ed19b58956b0be";
+  src = fetchFromGitHub {
+    owner = "bryanpkc";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "JiddvTbuOysenrVWGUEyKSzpCF1PJaYWQUdz3FuLCdw=";
```
Comment (NickCao) [inline pkgs/tools/networking/corkscrew/default.nix:10]:
```suggestion
    hash = "JiddvTbuOysenrVWGUEyKSzpCF1PJaYWQUdz3FuLCdw=";
```

File: pkgs/tools/networking/corkscrew/default.nix [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+
+  nativeBuildInputs = [ automake autoconf ];
 
-  preConfigure = ''
-    ln -sf ${automake}/share/automake-*/config.sub config.sub
-    ln -sf ${automake}/share/automake-*/config.guess config.guess
+  buildPhase = ''
+    autoreconf --install
+    ./configure --prefix=$out
+    make
```
Comment (NickCao) [inline pkgs/tools/networking/corkscrew/default.nix:20]: Just use `autoreconfHook`

**Review by NickCao — COMMENTED** [review]
And please make your commit messages conform to CONTRIBUTING.md

File: pkgs/tools/networking/corkscrew/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/tools/networking/corkscrew/default.nix:10]: (`hash =` only works with the SRI format, so the hash also need to be changed.)

**Comment by JackSullivan** [comment]
With respect to having my commit messages conform to the guidelines, should I also rebase these existing messages into a single conformant one?

**Comment by NickCao** [comment]
> With respect to having my commit messages conform to the guidelines, should I also rebase these existing messages into a single conformant one?

Yes, or split the changes into individual commits as you see appropriate.

**Review by NickCao — APPROVED** [review]
diff LGTM, only need to fixup the commit history.

**Review by NickCao — APPROVED** [review]

**Pull request merged**